### ISSUE TITLE
Merge HalfInteger improvements upstream

### ISSUE THREAD

**Issue by jagot** [JuliaAtoms/AtomicLevels.jl#18]

**Comment by jagot** [comment]
I also found this problem with ranges of `HalfInteger`s:

```julia
julia> hi"1":hi"-1":hi"-1"
ERROR: rem not defined for HalfInteger
Stacktrace:
 [1] error(::String, ::String, ::Type) at ./error.jl:42
 [2] no_op_err(::String, ::Type) at ./promotion.jl:410
 [3] rem(::HalfInteger, ::HalfInteger) at ./promotion.jl:429
 [4] (::Colon)(::HalfInteger, ::HalfInteger, ::HalfInteger) at ./range.jl:207
 [5] top-level scope at none:0

julia> reverse(hi"-1":hi"1")
3-element Array{HalfInteger,1}:
  1
  0
 -1
```

**Comment by mortenpi** [comment]
I should probably implement the 3-argument version of `Base.:(:)` for completeness.

**Comment by jagot** [comment]
Implemented in WignerSymbols v0.3.0 and f7d5253. Pending checks.

**Comment by jagot** [comment]
Fixed! Thanks @mortenpi
